scheduler/test/cook/test/mesos/api.clj

```diff
@@ -1187,19 +1187,18 @@
   (with-redefs [dl/job-uuid->dataset-maps-cache (util/new-cache)]
     (let [expected-job-map
           (fn
-            ; Converts the provided job and framework-id (framework-id) to the job-map we expect to get back from
+            ; Converts the provided job to the job-map we expect to get back from
             ; api/fetch-job-map. Note that we don't include the submit_time field here, so assertions below
             ; will have to dissoc it.
             [{:keys [mem max-retries max-runtime expected-runtime name gpus
                      command ports priority uuid user cpus application
                      disable-mea-culpa-retries executor datasets]
-              :or {disable-mea-culpa-retries false}}
-             framework-id]
+              :or {disable-mea-culpa-retries false}}]
             (cond-> {;; Fields we will fill in from the provided args:
                      :command command
                      :cpus cpus
                      :disable_mea_culpa_retries disable-mea-culpa-retries
-                     :framework_id framework-id
+                     :framework_id nil
```

Comment:
Can this be removed instead of being set to `nil`?

Reply:
No. Unit tests use this to make the pre-canned response they validate against. I either have the field, and set it to nil, or I have to wrap a bunch of unit tests with (with-redefs [framework-id-config (fn .....].

Reply:
Did the redef as we discussed.